File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_sentences(string letters);
int count_letters(string letters);
int count_words(string letters);

int main(void)

{
    string text = get_string("Text: ");
    int count = count_letters(text);
    int c_words = count_words(text);
    int c_sentences = count_sentences(text);
    float l, s;
    l = count / (float) c_words * 100;
    s = c_sentences / (float) c_words * 100;
    int i =  round(0.0588 * l - 0.296 * s - 15.8);
    printf("%i || %i || %i", count, c_words, c_sentences);
    printf("%0.2f || %0.2f || %i", l, s, i);
    if (i < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (i > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", i);
    }
}
int count_letters(string letters)
{
    int s = strlen(letters);
    char character;
    int contador = 0;
    for (int i = 0; i < s ; i++)
    {
        character = letters[i];
        if ((character > 64 && character < 91) || (character >= 97 && character <= 122))
        {
            contador++;
        }
    }
    return contador;
}
int count_sentences(string letters)
{
    int s = strlen(letters);
    char character;
    int contador = 0;
    for (int i = 0; i < s ; i++)
    {
        character = letters[i];
        if (character == 33 || character == 46  || character == 63)
        {
            contador++;
        }
    }
    return contador;

}
int count_words(string letters)
{
    int s = strlen(letters);
    char character;
    int contador = 1;
    for (int i = 0; i < s ; i++)
    {
        character = letters[i];
        if (character == 32)
        {
            contador++;
        }
    }
    return contador;
}

File: sentimental-readability/readability.py
# TODO

def main():
    text = input("Text: ")
    count = count_letters(text)
    c_words = count_words(text)
    c_sentences = count_sentences(text)
    l = count / c_words * 100
    s = c_sentences / c_words * 100
    i = round(0.0588 * l - 0.296 * s - 15.8)
    print(f"{count} || {c_words} = {c_sentences}")
    if (i < 1):
        print("Before Grade 1\n")
    elif (i > 16):
        print("Grade 16+\n")
    else:
        print(f"Grade {i}\n")

def count_letters(letters):
    i = 0
    for letter in letters:
        if (letter.isalpha()):
            i+=1
    return i

def count_words(letters):
    i = 1
    for letter in letters:
        if(letter == " "):
            i+=1
        print(f"{letter}")
    return i

def count_sentences(letters):
    i = 0
    characters = (".", "!", "?")
    for letter in letters:
        if (letter in characters):
            i+=1

    return i
if __name__ == "__main__":
    main()
